dexmaker/src/main/java/com/android/dx/AppDataDirGuesser.java

```diff
@@ -179,4 +190,38 @@ boolean fileOrDirExists(File file) {
     boolean isWriteableDirectory(File file) {
         return file.isDirectory() && file.canWrite();
     }
+
+    Integer getProcessUid() {
+        /* Uses reflection to try to fetch process UID. It will only work when executing on
+         * Android device. Otherwise, returns null.
+         */
+        try {
+            Method myUid = Class.forName("android.os.Process").getMethod("myUid");
```

Comment:
I guess we cannot use https://developer.android.com/reference/android/content/Context.html#getDataDir() as this only works for API24 and later

Reply:
Correct, multi-user was introduced in JellyBean MR1 - Android 4.2. Process.myUid is from API 2.